inference-engine/src/cldnn_engine/cldnn_infer_request.cpp

```diff
@@ -958,77 +925,125 @@ void copyToFloat(float* dst, const InferenceEngine::Blob* src) {
 
 }  // namespace
 
-void CLDNNInferRequest::PrepareInput(const cldnn::primitive_id &inputName, const Blob &inputBlob) {
-    OV_ITT_SCOPED_TASK(itt::domains::CLDNNPlugin, "CLDNNInferRequest::PrepareInput");
-    // Get input layout
-    if (m_graph->GetInputLayouts().find(inputName) == m_graph->GetInputLayouts().end()) {
-        IE_THROW() << "Input name mismatch.";
+template <typename TSRC, typename TDST>
+void copyToMem(cldnn::memory& mem, const InferenceEngine::Blob* src) {
+    OV_ITT_SCOPED_TASK(itt::domains::CLDNNPlugin, "copyToMem");
+
+    const InferenceEngine::TBlob<TSRC>* t_blob = dynamic_cast<const InferenceEngine::TBlob<TSRC>*>(src);
+    if (t_blob == nullptr) {
+        IE_THROW() << "input type is " << src->getTensorDesc().getPrecision() << " but input is not "
+            << typeid(TSRC).name();
     }
-    auto inputLayout = m_graph->GetInputLayouts().at(inputName);
-    auto is_same_buffer = [](const Blob& blob, const cldnn::memory& memory) -> bool {
-        const std::string str_not_allocated("Input data was not allocated.");
-        cldnn::pointer<const uint8_t> ptr = memory.pointer<const uint8_t>();
-        const uint8_t* blob_ptr = blob.cbuffer().as<const uint8_t*>();
-        const uint8_t* mem_ptr = ptr.data();
-        if (blob_ptr == nullptr || mem_ptr == nullptr) {
-            IE_THROW() << str_not_allocated;
-        }
-        return (blob_ptr == mem_ptr) && (blob.byteSize() == memory.size());
-    };
 
-    cldnn::primitive_id internalName = "parameter:" + inputName;
-    const cldnn::memory& memory = inputsMemory.at(inputName);
+    auto ptr = mem.pointer<TDST>();
+    TDST* dst = ptr.data();
+    if (!dst) {
+        IE_THROW() << "invalid memory object";
+    }
+    const TSRC* srcPtr = t_blob->readOnly();
+    if (srcPtr == nullptr) {
+        IE_THROW() << "Input data was not allocated.";
+    }
+    for (size_t i = 0; i < t_blob->size(); i++) dst[i] = srcPtr[i];
```

Comment:
nit: please avoid writing loop body on the same line with loop definition. 
```cpp
for (size_t i = 0; i < t_blob->size(); i++) 
    dst[i] = srcPtr[i];
```
is easier to read

Reply:
Fixed